### ISSUE TITLE
OutOfMemoryError in ExplodeJarTask

### ISSUE THREAD

**Issue by daanschipper** [autonomousapps/dependency-analysis-android-gradle-plugin#823]
**Build scan link**
Private, can't share unfortunately.

**Plugin version**
`v1.13.1`, however no changes in `ExplodeJarTask` in ` main..v1.13.1`

**Gradle version**
`7.4.2`

**(Optional) Android Gradle Plugin (AGP) version**
Not applicable.

**`reason` output for bugs relating to incorrect advice**
Not applicable.

**Describe the bug**
When serializing the output of `ExplodeJarTask` to file, it sometimes throws an `OutOfMemoryError`:
```
Caused by: java.lang.OutOfMemoryError: Java heap space
at okio.Buffer.readString(Buffer.kt:313)
at okio.Buffer.readUtf8(Buffer.kt:297)
at com.squareup.moshi.JsonAdapter.toJson(JsonAdapter.java:94)
at com.autonomousapps.tasks.ExplodeJarTask$ExplodeJarWorkAction.execute(ExplodeJarTask.kt:126)
```

Specifically the call `explodedJars.toJson()` causes this. To give some context, `explodedJars.toJson().length = 5385952` in our case. The daemon is configured to run with 8Gb already.

**To Reproduce**
It's not happening deterministically unfortunately,  but having an insane number of dependencies sometimes triggers this.

**Expected behavior**
No OOM exception.

**Additional context**
From some initial investigation the entire `Set<ExplodedJar>` is first serialized to json and then written to file. Maybe it makes more sense to serialize the `ExplodedJar` one by one?

**Comment by daanschipper** [comment]
This is also similarly happening in `SynthesizeDependenciesTask` where the entire object is first converted to `String` and then written to the output file
```
Caused by: java.lang.OutOfMemoryError: Java heap space
at kotlin.io.TextStreamsKt.copyTo(ReadWrite.kt:125)
at kotlin.io.TextStreamsKt.copyTo$default(ReadWrite.kt:120)
at kotlin.io.TextStreamsKt.readText(ReadWrite.kt:107)
at kotlin.io.FilesKt__FileReadWriteKt.readText(FileReadWrite.kt:125)
at kotlin.io.FilesKt__FileReadWriteKt.readText$default(FileReadWrite.kt:125)
at com.autonomousapps.tasks.SynthesizeDependenciesTask$SynthesizeDependenciesWorkAction.execute(SynthesizeDependenciesTask.kt:234)
```

**Comment by daanschipper** [comment]
@autonomousapps any chance you could release a new version with this fix? We are quite heavily impacted by this.

**Comment by autonomousapps** [comment]
No prob. Just released it as 1.17.0 💪
